Added ComboBox Filtering and Localization (#902)
telerik/maui-docs: controls/combobox/filtering.md

---
title: Filtering
page_title: .NET MAUI ComboBox Documentation - Filtering
description: Learn more about the filtering functionalities in Telerik UI for .NET MAUI ComboBox control.
position: 5
slug: combobox-filtering
---

# .NET MAUI ComboBox Filtering

The [Telerik UI for .NET MAUI ComboBox]({%slug combobox-overview%}) provides filtering, which allows the users to refine their search results as they type into the input field. The default filtering is performed according to the `SearchMode` and `SearchTexhPath` properties of the ComboBox. 

> For more details on the search functionality of the ComboBox, go to [Edit Mode & Search]({%slug combobox-editmode-and-search%}) topic.

Reviewer: ```suggestion
For more details on the search functionality of the ComboBox, go to the [Edit Mode & Search]({%slug combobox-editmode-and-search%}) topic.
```

> This topic describes in details the configuration options related to the filtering feature the ComboBox:

Reviewer: ```suggestion
This topic describes in detail the configuration options related to the filtering feature of the ComboBox:
```

* `IsFilteringEnabled`(`bool`)&mdash;Indicates whether the filtering is enabled. Its default value is `False`.
* `FilteredItems`(`IReadOnlyCollection<object>`)&mdash;Gets the filtered items collection.

> * `NoResultsMessage`(`string`)&mdash;Defines the message visualized when there are no matched items found during filtering.

Reviewer: ```suggestion
* `NoResultsMessage`(`string`)&mdash;Defines the message visualized when no items match the applied filter.
```

> * `NoResultsTemplate`(`DataTemplate`)&mdash;Defines the `DataTemplate` visualized when there are no matched items found during filtering.

Reviewer: ```suggestion
* `NoResultsTemplate`(`DataTemplate`)&mdash;Defines the `DataTemplate` visualized when no items match the applied filter.
```

Here is a quick example of `RadComboBox` with enabled filtering:

**1.** First, add the ComboBox definition with `IsEditable`, `IsFilteringEnabled` and `SearchMode` applied:

<snippet id='combobox-filtering' />

**2.** Add the `ViewModel` class:

<snippet id='combobox-cities-viewmodel' />

**3.** Add the `City` data item:

<snippet id='combobox-city-businessmodel' />

Here is the result:

![Telerik .NET MAUI ComboBox Filtering](images/combobox-filtering.gif)

## Custom Filtering Behavior

> You can override the default logic for filtering items by applying a custom filtering behavior to the ComboBox. In this case the built-in search is performed upon the filtered items. Create a custom class that inherits from `ComboBoxFilteringBehavior` and override its `FilterItems` method, then assign it to the `ComboBox`'s `FilteringBehavior` property.

Reviewer: ```suggestion
You can override the default item filtering logic by applying custom filtering in the ComboBox. The custom filtering feature uses the built-in search functionality under the hood. To implement a custom filter:
1. Create a custom class that inherits from `ComboBoxFilteringBehavior`.
1. Override the `FilterItems` method of your custom class.
1. Assign the `ComboBoxFilteringBehavior` to the `FilteringBehavior` property of the `ComboBox`.
```

* `FilteringBehavior`(`ComboBoxFilteringBehavior`)&mdash;Defines the filtering behavior used to filter items.

> Here is a quick example which demonstrates a custom filtering behavior - the items are filtered not only by the entered search text, but also according to the `Population` property. 

Reviewer: ```suggestion
Here is a quick example which demonstrates the custom filtering behavior—the items are filtered not only by the entered search text, but also according to the `Population` property. 
```
